skimage/measure/fit.py

```diff
@@ -785,54 +790,64 @@ def ransac(data, model_class, min_samples, residual_threshold,
 
     random_state = check_random_state(random_state)
 
-    if min_samples < 0:
-        raise ValueError("`min_samples` must be greater than zero")
+    # in case data is not pair of input and output, male it like it
+    if not isinstance(data, (tuple, list)):
+        data = (data, )
+    data_lens = list(map(len, data))
+    assert len(set(data_lens)) == 1, 'all data have to have the same number of samples,' \
+                                     ' provided: %r' % [d.shape for d in data]
+    num_samples = data_lens[0]
+
+    if not (0 < min_samples <= num_samples):
+        raise ValueError("`min_samples` must be in range (0, <number-of-samples>]")
+
+    if residual_threshold < 0:
+        raise ValueError("`residual_threshold` must be greater than zero")
 
     if max_trials < 0:
         raise ValueError("`max_trials` must be greater than zero")
 
-    if stop_probability < 0 or stop_probability > 1:
-        raise ValueError("`stop_probability` must be in range [0, 1]")
+    if not (0 <= stop_probability <= 1):
+        raise ValueError("`stop_probability` must be in range (0, 1)")
```

Comment:
The brackets should not be changed here?

Reply:
sorry, my confusion with open and close intervals... 
![image](https://user-images.githubusercontent.com/6035284/61992457-0abc3900-b05f-11e9-9024-934282eac942.png)